### PR TITLE
Config validation clean up

### PR DESCRIPTION
**Pull request by awanlin**
## Hey, I just made a Pull Request!

While validating the various issues fixed by PR #18837 I noticed that there was several items that were coming up as invalid when running: `./packages/cli/bin/backstage-cli config:check --config app-config.yaml --lax --strict`:

```text
Error: Configuration does not match schema

  Config must NOT have additional properties { additionalProperty=pagerduty } at
  Config must NOT have additional properties { additionalProperty=authorizationUrl } at /auth/providers/oidc/development
  Config must NOT have additional properties { additionalProperty=tokenUrl } at /auth/providers/oidc/development
  Config must NOT have additional properties { additionalProperty=tokenEndpointAuthMethod } at /auth/providers/oidc/development
  Config must NOT have additional properties { additionalProperty=tokenSignedResponseAlg } at /auth/providers/oidc/development
  Config must NOT have additional properties { additionalProperty=host } at /integrations/awsS3/0
```

This PR address these invalid items and corrects the `app-config.yaml` or the related `config.d.ts` file

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: app-config.yaml [hunk 1]
```diff
@@ -211,7 +211,7 @@ integrations:
   #    clientEmail: 'example@example.com'
   #    privateKey: ${GCS_PRIVATE_KEY}
   awsS3:
-    - host: amazonaws.com
+    - endpoint: ${AWS_S3_ENDPOINT}
```
Comment (awanlin) [inline app-config.yaml:214]: There is no `host` property in the `aws3` config - https://github.com/backstage/backstage/blob/1a14adeb574f7846b6d5f5ef555a57dea403e03a/packages/integration/config.d.ts#L253 - replace with `endpoint`

File: app-config.yaml [hunk 2]
```diff
@@ -352,8 +352,6 @@ auth:
         metadataUrl: ${AUTH_OIDC_METADATA_URL}
         clientId: ${AUTH_OIDC_CLIENT_ID}
         clientSecret: ${AUTH_OIDC_CLIENT_SECRET}
-        authorizationUrl: ${AUTH_OIDC_AUTH_URL}
-        tokenUrl: ${AUTH_OIDC_TOKEN_URL}
```
Comment (awanlin) [inline app-config.yaml:356]: I could not find these any where, if they were used I belive they would have been mentioned here: https://github.com/backstage/backstage/blob/1a14adeb574f7846b6d5f5ef555a57dea403e03a/plugins/auth-backend/src/providers/oidc/provider.ts#L231

File: app-config.yaml [hunk 3]
```diff
@@ -434,8 +432,10 @@ costInsights:
       kind: 'PINTS_OF_ICE_CREAM'
       unit: 'ice cream pint'
       rate: 5.5
-pagerduty:
+
+pagerDuty:
   eventsBaseUrl: 'https://events.pagerduty.com/v2'
```
Comment (awanlin) [inline app-config.yaml:437]: Updated with capital "D" to match config: https://github.com/backstage/backstage/blob/1a14adeb574f7846b6d5f5ef555a57dea403e03a/plugins/pagerduty/config.d.ts#L21

File: plugins/auth-backend/config.d.ts [hunk 4]
```diff
@@ -152,6 +152,8 @@ export interface Config {
           clientSecret: string;
           callbackUrl?: string;
           metadataUrl: string;
+          tokenEndpointAuthMethod?: string;
+          tokenSignedResponseAlg?: string;
```
Comment (awanlin) [inline plugins/auth-backend/config.d.ts:156]: Updated to match config here: https://github.com/backstage/backstage/blob/1a14adeb574f7846b6d5f5ef555a57dea403e03a/plugins/auth-backend/src/providers/oidc/provider.ts#L239-L244

**Comment by awanlin** [comment]
Hi @freben, would it make sense to run `yarn backstage-cli config:check --config app-config.yaml --lax --strict` as part of the CI to keep this up to date?

**Review by freben — APPROVED** [review]
Awesome! Let's get this in before the release.

**Comment by awanlin** [comment]
Thanks @freben! 🚀

**Pull request merged**